xocto/ranges.py

```diff
@@ -790,6 +790,21 @@ def intersection(self, other: Range[datetime.date]) -> Optional["FiniteDateRange
         assert base_intersection.boundaries == RangeBoundaries.INCLUSIVE_INCLUSIVE
         return FiniteDateRange(base_intersection.start, base_intersection.end)
 
+    def is_disjoint(self, other: Range[datetime.date]) -> bool:
+        # Adjacent dates should not be considered disjoint, we extend the other
+        # range to allow them to be considered adjacent.
+        other_start = other.start
+        if other._is_right_inclusive:
+            assert other.start
```

Comment:
❓ Why does this assertion hold? I would expect `(None, '2022-01-01']` to be a valid range.

Maybe this should be `._is_left_inclusive`?

Reply:
🤔  You are right, this is a bug, I will review my tests to see why they weren't failing.